File: src/data/courses/dsa/essential-math/basic-game-problems/nim-game/code/python/nim_game.py
def nim_game(piles): xor_sum=0; [xor_sum:=xor_sum^p for p in piles]; return xor_sum!=0
def can_win_stones(n): return n%3!=0
print(f"Nim [3,4,5]: {nim_game([3,4,5])}"); print(f"Can win with 4 stones: {can_win_stones(4)}")

File: src/data/courses/dsa/essential-math/basic-game-problems/nim-game/code/java/NimGame.java
public class NimGame {
    boolean nimGame(int[] piles) { int xor=0; for(int p:piles) xor^=p; return xor!=0; }
    boolean canWinStones(int n) { return n%3!=0; }
    public static void main(String[] a) { NimGame g=new NimGame(); System.out.println("Nim [3,4,5]: "+g.nimGame(new int[]{3,4,5})); System.out.println("Can win with 4 stones: "+g.canWinStones(4)); }
}